app/src/main/java/io/gnosis/safe/ui/transaction/TransactionViewHolderFactory.kt

```diff
@@ -36,87 +56,223 @@ class TransactionViewHolderFactory : BaseFactory<BaseTransactionViewHolder<Trans
-    BaseTransactionViewHolder<TransactionView.ChangeMastercopy>(viewBinding) {
+private const val OPACITY_FULL = 1.0F
 
-    override fun bind(data: TransactionView.ChangeMastercopy, payloads: List<Any>) {
-        TODO("Not yet implemented")
+class TransferViewHolder(private val viewBinding: ItemTxTransferBinding) :
+    BaseTransactionViewHolder<TransactionView.Transfer>(viewBinding) {
+    private val resources = viewBinding.root.context.resources
+
+    override fun bind(viewTransfer: TransactionView.Transfer, payloads: List<Any>) {
+        with(viewBinding) {
+            finalStatus.setText(viewTransfer.statusText)
+            finalStatus.setTextColor(resources.getColor(viewTransfer.statusColorRes))
+            amount.text = viewTransfer.amountText
+            amount.setTextColor(resources.getColor(viewTransfer.amountColor))
+            dateTime.text = viewTransfer.dateTimeText
+            txTypeIcon.setImageResource(viewTransfer.txTypeIcon)
+            blockies.setAddress(viewTransfer.address)
+            ellipsizedAddress.text = viewTransfer.address.formatForTxList()
+
+            finalStatus.alpha = OPACITY_FULL
+            amount.alpha = viewTransfer.alpha
+            dateTime.alpha = viewTransfer.alpha
+            txTypeIcon.alpha = viewTransfer.alpha
+            blockies.alpha = viewTransfer.alpha
+            ellipsizedAddress.alpha = viewTransfer.alpha
+        }
+    }
+}
+
+class TransferQueuedViewHolder(private val viewBinding: ItemTxQueuedTransferBinding) :
+    BaseTransactionViewHolder<TransactionView.TransferQueued>(viewBinding) {
+    private val resources = viewBinding.root.context.resources
+
+    override fun bind(viewTransfer: TransactionView.TransferQueued, payloads: List<Any>) {
+        with(viewBinding) {
+            status.setText(viewTransfer.statusText)
+            status.setTextColor(resources.getColor(viewTransfer.statusColorRes))
+            amount.text = viewTransfer.amountText
+            dateTime.text = viewTransfer.dateTimeText
+            txTypeIcon.setImageResource(viewTransfer.txTypeIcon)
+            blockies.setAddress(viewTransfer.address)
+            ellipsizedAddress.text = viewTransfer.address.formatForTxList()
+            amount.setTextColor(resources.getColor(viewTransfer.amountColor))
+            confirmations.setTextColor(resources.getColor(viewTransfer.confirmationsTextColor))
+            confirmationsIcon.setImageDrawable(resources.getDrawable(viewTransfer.confirmationsIcon))
+            confirmations.text = resources.getString(R.string.tx_list_confirmations, viewTransfer.confirmations, viewTransfer.threshold)
+            nonce.text = viewTransfer.nonce
+        }
     }
 }
 
 class SettingsChangeViewHolder(private val viewBinding: ItemTxSettingsChangeBinding) :
     BaseTransactionViewHolder<TransactionView.SettingsChange>(viewBinding) {
+    private val resources = viewBinding.root.context.resources
```

Comment:
let's not keep  reference to `context` related objects in the `viewHolder`. If you need `resources` somewhere in your `bind` method, I would ask you to inline this `val` or move it within the `bind` method. You have access to `viewBinding` there anyhow.

Reply:
Moved to bind method